test/extended/networking/egressip.go

```diff
@@ -0,0 +1,654 @@
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By(fmt.Sprintf("Sending requests from prober and making sure that %d requests with search string and EgressIPs %v were seen", iterations, egressIPSet))
+				sendEgressIPProbesAndCheckPacketSnifferLogs(oc, externalNamespace, probePodName, routeName, targetProtocol, targetHost, targetPort, iterations, iterations, packetSnifferDaemonSet, egressIPSet)
+			})
+
+			g.It("only pods matched by the pod selector should have the EgressIPs", func() {
+				g.By("Creating the EgressIP test source deployment with number of pods equals number of EgressIP nodes")
+				deployment0Name, route0Name, err := createAgnhostDeploymentAndIngressRoute(oc, egressIPNamespace, "0", ingressDomain, len(egressIPNodesOrderedStr), egressIPNodesOrderedStr)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By("Creating the second EgressIP test source deployment with number of pods equals number of EgressIP nodes")
+				_, route1Name, err := createAgnhostDeploymentAndIngressRoute(oc, egressIPNamespace, "1", ingressDomain, len(egressIPNodesOrderedStr), egressIPNodesOrderedStr)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				// For this test, get a single EgressIP per node.
+				// Note: On some clouds like GCP, there is no dedicated CIDR per node and instead all EgressIPs come from a common pool.
+				// Thus, this is only an artificial assignment of EgressIP to node on these cloud platforms and the EgressIP feature
+				// will pick the actual node.
+				g.By("Getting a map of source nodes and potential Egress IPs for these nodes")
+				egressIPsPerNode := 1
+				nodeEgressIPMap, err := findNodeEgressIPs(oc, egressIPNodesOrderedStr, cloudType, egressIPsPerNode)
+				framework.Logf("%v", nodeEgressIPMap)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By("Choosing the EgressIPs to be assigned, one per node")
+				var egressIPs []string
+				egressIPSet := make(map[string]struct{})
+				for _, eip := range nodeEgressIPMap {
+					_, ok := egressIPSet[eip[0]]
+					if !ok {
+						egressIPs = append(egressIPs, eip[0])
+						egressIPSet[eip[0]] = struct{}{}
+					}
+				}
+
+				g.By("Marshalling the desired EgressIPs into a string")
+				egressIPsString, err := json.Marshal(egressIPs)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By("Creating the EgressIP object and writing it to disk")
+				var egressIPConfig = fmt.Sprintf(
+					egressIPYamlTemplatePodAndNamespaceSelector, // template yaml
+					egressIPNamespace,                       // name of EgressIP
+					egressIPsString,                         // compact yaml of egressIPs
+					fmt.Sprintf("app: %s", deployment0Name), // pod selector
+					fmt.Sprintf("kubernetes.io/metadata.name: %s", egressIPNamespace), // namespace selector
+				)
+				err = ioutil.WriteFile(tmpDirEgressIP+"/"+egressIPYaml, []byte(egressIPConfig), 0644)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				// Run this twice to make sure that repeated EgressIP creation and deletion works.
```

Comment:
don't see this happening twice. copy/paste from above?

Reply:
yep, removed